mixer/pkg/protobuf/yaml/walker.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yaml
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+)
+
+// Visitor defines the callback made by the Walker.
+type Visitor interface {
+	Visit(name string, val interface{}, field *descriptor.FieldDescriptorProto) (visitor Visitor,
```

Comment:
Why do you need the returned visitor? It looks like you don't need this?

Reply:
return `Visitor` lets us propagate context if needed. This particular PR does not need to propagate context (perhaps...), but while building a tree message generators, a `Visit()` call may return a child `Visitor`.

Reply:
Not applicable anymore